### PR TITLE
No more getting stuck in a Dizzy Mushroom-Hole

### PR DESCRIPTION
**Pull request by Therandomhoboo**
# Document the changes in your pull request

Whoever put in Hallucinogen Mushrooms & Hippie's delight without sorting out the code should be burned at a stake. I've become fully grey and aged fifty years. However, it is now fixed. I am finally a coder.

# Why is this good for the game?
Gives admins less of an headache, gives players the ability to not throw up from having a constant dizzy effect and makes me happy because I don't have to fix this anymore.

# Testing

Yes, I tested this by drinking a ton of Hippie's delight & Halluc-Mushroom and waiting for it to run out. You're welcome.

# Changelog

:cl:  

bugfix: Hallucinogen Mushroom & Hippie's delight no longer perma dizzy you
tweak: Hippie's delight numbers for how long their effects last (dizzy, druggy, jitter etc) have been sorted out (Thanks to @Moltijoe)

/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -1509,26 +1509,26 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	M.set_slurring_if_lower(1 SECONDS * REM)
 	switch(current_cycle)
 		if(1 to 5)
-			M.adjust_dizzy(10)
-			M.set_drugginess(30)
+			M.set_dizzy_if_lower(10 SECONDS)
+			M.set_drugginess_if_lower(30 SECONDS)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]:
```suggestion
			M.set_dizzy_if_lower(1 SECONDS)
			M.set_drugginess_if_lower(3 SECONDS)
```

Comment (Moltijoe) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: undocumented change otherwise

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -1509,26 +1509,26 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
 			if(prob(10))
 				M.emote(pick("twitch","giggle"))
-		if(5 to 10)
-			M.adjust_jitter(20 SECONDS)
-			M.adjust_dizzy(20)
-			M.set_drugginess(45)
+		if(6 to 10)
+			M.set_jitter_if_lower(20 SECONDS)
+			M.set_dizzy_if_lower(20 SECONDS)
+			M.set_drugginess_if_lower(45 SECONDS)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]:
```suggestion
			M.set_dizzy_if_lower(2 SECONDS)
			M.set_drugginess_if_lower(4.5 SECONDS)
```

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 3]
```diff
@@ -1509,26 +1509,26 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
 			if(prob(20))
 				M.emote(pick("twitch","giggle"))
-		if (10 to 200)
-			M.adjust_jitter(40 SECONDS)
-			M.adjust_dizzy(40)
-			M.set_drugginess(60)
+		if (11 to 200)
+			M.set_jitter_if_lower(40 SECONDS)
+			M.set_dizzy_if_lower(40 SECONDS)
+			M.set_drugginess_if_lower(60 SECONDS)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]:
```suggestion
			M.set_dizzy_if_lower(4 SECONDS)
			M.set_drugginess_if_lower(6 SECONDS)
```

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 4]
```diff
@@ -1509,26 +1509,26 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
 			if(prob(30))
 				M.emote(pick("twitch","giggle"))
-		if(200 to INFINITY)
-			M.adjust_jitter(1 MINUTES)
-			M.adjust_dizzy(60)
-			M.set_drugginess(75)
+		if(201 to INFINITY)
+			M.set_jitter_if_lower(60 SECONDS)
+			M.set_dizzy_if_lower(60 SECONDS)
+			M.set_drugginess_if_lower(75 SECONDS)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]:
```suggestion
			M.set_dizzy_if_lower(6 SECONDS)
			M.set_drugginess_if_lower(7.5 SECONDS)
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]
these too

**Comment by Therandomhoboo** [comment]
> these too

Done, thank you.

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**